Mathlib/Topology/UniformSpace/Cauchy.lean

```diff
@@ -673,6 +673,15 @@ theorem isCompact_of_totallyBounded_isClosed [CompleteSpace α] {s : Set α} (ht
     (hc : IsClosed s) : IsCompact s :=
   (@isCompact_iff_totallyBounded_isComplete α _ s).2 ⟨ht, hc.isComplete⟩
 
+theorem isCompact_closure_of_totallyBounded_of_isComplete_closure {s : Set α}
+    (ht : TotallyBounded s) (hc : IsComplete (closure s)) : IsCompact (closure s) :=
+  isCompact_iff_totallyBounded_isComplete.mpr ⟨TotallyBounded.closure ht, hc⟩
+
+theorem isCompact_closure_of_totallyBounded_complete [CompleteSpace α] {s : Set α}
+    (ht : TotallyBounded s) : IsCompact (closure s) :=
+  isCompact_closure_of_totallyBounded_of_isComplete_closure ht
+    (IsClosed.isComplete isClosed_closure)
```

Comment:
These two lemmas seem a bit over-specialized to me, especially since you can just write `ht.closure` whenever `ht : TotallyBounded t`. Instead, I would suggest adding `TotallyBounded.isCompact_of_isComplete` and renaming/creating an alias for `isCompact_of_totallyBounded_isClosed` to `TotallyBounded.isCompact_of_isClosed`.

Then I think you don't really need the specialized closure lemmas: for example, the first one would just be `ht.closure.isCompact_of_isComplete hc`.